server/player/player.go

```diff
@@ -89,6 +89,11 @@ type Player struct {
 	lastXPPickup atomic.Value[time.Time]
 	immunity     atomic.Value[time.Time]
 
+	deathMu        sync.Mutex
+	hasDied        bool
+	deathPos       mgl64.Vec3
+	deathDimension world.Dimension
```

Comment:
Would just make one of these fields a pointer so you can remove the `deathPos` field.

Reply:
Did you mean the `hasDied` field?

Reply:
I did, sorry!

Reply:
Removed.